### PR TITLE
fix(pluck): add object kind

### LINKED ISSUE #253: pluck typings issue
Hello!

I have the latest typings (installed as `yarn add --dev types/npm-ramda#dist`).
 
My code looks approximately like this:

```
R.pluck('value',
  R.pick(['a'])({
    'a': { value: 'some_value' },
    'b': { value: 'some_value' },
  }) // = {'a': {value: 'some_value'}}
)
```

Typescript throws:
```
TS2345:
Argument of type 'Pick<{ 'a': { value: string; }; 'b': { value: string; }; }, "a">' is not assignable to parameter of type 'List<any>'.

Type 'Pick<{ 'a': { value: string; }; 'b': { value: string; }; }, "a">' is not assignable to type 'ArrayLike<any>'.

Property 'length' is missing in type 'Pick<{ 'a': { value: string; }; 'b': { value: string; }; }, "a">'.
```

First of all such use case is valid accordingly to documentation:
```
R.pluck('a')([{a: 1}, {a: 2}]); //=> [1, 2]
R.pluck(0)([[1, 2], [3, 4]]);   //=> [1, 3]
R.pluck('val', {a: {val: 3}, b: {val: 5}}); // << My case. Typescript also throws here. 
```
Possibly I misunderstand something, but from my point of view, this looks like missing typings for dictionary-like argument type. Thank you!

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #253

### CODE REVIEW

**Comment by bloadvenro** [comment]
I've cloned `fix/issue-253` and tried to check a couple of moments.

`R_pluck('a')(x_y_z_with_a_1_record);` - this curried version doesn't seem to work:

```
file: 'file:///.../npm-ramda/tests/pluck.ts'

'Argument of type '{ x: { a: 1; }; y: { a: true; }; z: { a: "string"; }; }' is not assignable to parameter of type 'List<Record<"a", {}>>'.

Type '{ x: { a: 1; }; y: { a: true; }; z: { a: "string"; }; }' is not assignable to type 'ArrayLike<Record<"a", {}>>'.

Property 'length' is missing in type '{ x: { a: 1; }; y: { a: true; }; z: { a: "string"; }; }'.'
at: '28,14'
```

Also if I try

```
declare const x_y_z_with_a_1_record: {
  x: { a: 1 };
  y: { a: true };
  z: { a: 'string' };
};
```
the result type is
```
{
  x:  true | "string" | 1,
  y:  true | "string" | 1,
  z: `true | "string" | 1
}
```
but I know it is possible to map value types as `{ x:  number, y:  boolean, z: string }`.

Actually, I was able to solve both problems by manually editing ramda definitions in my project:

```
import {
   List,
+Dictionary,
   Property
} from "./$types";

...

+<K extends string, T extends Dictionary<Dictionary<any>>>(key: K, obj: T): {[P in keyof T]: T[P][K]};

...

type pluck_record_10<K extends string> = {
    <V, T extends Record<K, V>>(list: List<T>): pluck_record_11<K, V, T>;
  +<T extends Dictionary<Dictionary<any>>>(obj: T): {[P in keyof T]: T[P][K]};
};
```

As a result we have:

```
R.pluck('val', {a: {val: 3}, b: {val: 'x'}}).a.toExponential; // OK, is a number
R.pluck('val', {a: {val: 3}, b: {val: 'x'}}).b.toUpperCase; // OK, is a string
R.pluck('val', {a: {val: 3}, b: {val: 'x'}}).a.toExponential; //OK, is a number
R.pluck('val')({a: {val: 3}, b: {val: 'x'}}).b.toUpperCase; //OK, is a string
R.pluck('a')([{a: 1}, {a: 2}]).length; // OK, is an array
R.pluck(0)([[1, 2], [3, 4]]).length;   // OK, is an array
```

**Comment by bloadvenro** [comment]
A apologize, but pull request seemed to me a difficult task)

**Comment by ikatyang** [comment]
> ```diff
> type pluck_record_10<K extends string> = {
>   <V, T extends Record<K, V>>(list: List<T>): pluck_record_11<K, V, T>;
> +  <T extends Dictionary<Dictionary<any>>>(obj: T): {[P in keyof T]: T[P][K]};
> };
> ```

This one is currently not available since I haven't thought that case while implementing [`dts-element-fp`](https://github.com/ikatyang/dts-element-fp), it only supports all-independent case and all-combinable case, no support for mixed case, opened https://github.com/ikatyang/dts-element-fp/issues/56 to track this issue.

File: tests/ramda-tests.ts [hunk 1]
```diff
@@ -2285,6 +2285,14 @@ import * as R from '../ramda/dist/index';
   R.pluck('a')([{ a: 1 }, { a: 2 }]); //=> [1, 2]
   // @dts-jest:pass:snap
   R.pluck(0)([[1, 2], [3, 4]]); //=> [1, 3]
+  // @dts-jest:pass:snap
+  R.pluck(
```
Comment (tycho01) [inline tests/ramda-tests.ts]: Maybe (also) test without `pick`?

**Pull request merged**